lib/dao.js

```diff
@@ -2687,6 +2687,14 @@ DataAccessObject.replaceById = function(id, data, options, cb) {
 
   Model.notifyObserversOf('before save', context, function(err, ctx) {
     if (err) return cb(err);
+    if (Model.settings.forceId && ctx.instance[pkName] !== id) {
+      var err = new Error ('id property cannot be changed from ' +
+        id + ' to ' + inst[pkName] + ' in \'before save\' operation ' +
+        'hook when forceId is set to true');
```

Comment:
@bajtos Very good question! I don't know why I didn't see this from a better perspective; in DBs you can never change PK irrespective of whether you set `forceId` or not. also it merits consideration this has been handled under the hood in my implemented methods in connectors; please see how we handle this situation in [loopback-connector for all relational DB](https://github.com/strongloop/loopback-connector/blob/958c6619e3353c4604adc15161fa3a39c8767dfd/lib/sql.js#L617), [memory-connector](https://github.com/strongloop/loopback-datasource-juggler/blob/8f347fbacb3020a86c186ef6e3ec8e0f0a9705db/lib/connectors/memory.js#L745).

I'll add tests to confirm this. So now I believe we have two options:

* Ignore `id` which is passed in `before save` hook, but dont throw an error, just give a warning with an appropriate message saying `the ID, X, which has been passed in before save hook has been ignored. You cannot change ID in replaceById method`

Or
* Throw an error and do not call `connector.replaceById()` anymore

Which one do you think is more appropriate in this case?